<p>
  Each case statement should have a default case. Additionally, the default case should fail the catalog compilation when the resulting behavior cannot be predicted on the majority of platforms the module will be used on. If you want the default case to be “do nothing,” include it as an explicit <code>default: {}</code> with a comment for clarity’s sake.
</p>

<h2>Noncompliant Code Example</h2>

<pre>
case $::operatingsystem {
  'centos': {
    $version = '1.2.3'
  }
  'solaris': {
    $version = '3.2.1'
  }
}
</pre>

<h2>Compliant Solution</h2>

<pre>
case $::operatingsystem {
  'centos': {
    $version = '1.2.3'
  }
  'solaris': {
    $version = '3.2.1'
  }
  default: {
    fail("Module ${module_name} is not supported on ${::operatingsystem}")
  }
}

case $::operatingsystem {
  'centos': {
    $version = '1.2.3'
  }
  'solaris': {
    $version = '3.2.1'
  }
  default: {
    # Do nothing because...
  }
}
</pre>
<h2>See</h2>
<ul>
  <li><a href="http://docs.puppetlabs.com/guides/style_guide.html#defaults-for-case-statements-and-selectors">See Puppet Labs Puppet Language Style Guide</a></li>
</ul>
