<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all classes and defines match the following regular expression: <nobr><code>^(::)?([a-z][a-z0-9_]*::)*[a-z][a-z0-9_]*$</code></nobr></p>
<h2>Noncompliant Code Example</h2>
<pre>
class my-class {}    # Noncompliant
define myDefine {}   # Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
class my_class {}
define my_define {}
</pre>
