<p>For readability reasons, the same starting token should be used for each comment. This rule checks that each comment starts with <code>#</code>.</p>
<h2>Noncompliant Code Example</h2>
<pre skipautotest="">
// My comments...
notice('hello') // My comments...

/* My comments... */
notice('hello') /* My comments... */
</pre>
<h2>Compliant Solution</h2>
<pre>
# My comments...
notice('hello') # My comments...
</pre>
<h2>See</h2>
<ul>
    <li><a href="https://docs.puppetlabs.com/guides/style_guide.html#comments">See Puppet Labs Puppet Language Style Guide</a></li>
</ul>



