<p>For readability reasons, the same starting token should be used for each comment. This rule checks that each comment starts with <code>#</code>.</p>
<h2>Noncompliant Code Example</h2>
<pre>
// My comments...
notice('hello') // My comments...

/* My comments... */
notice('hello') /* My comments... */
</pre>
<h2>Compliant Solution</h2>
<pre>
# My comments...
notice('hello') # My comments...
</pre>



