<p>Every class an define should be commented to explain its goal and how it works. This comment should be placed just
    before the class/define declaration.</p>

<h2>Noncompliant Code Example</h2>
<pre>
class configuration { ... }

define server() { ... }
</pre>

<h2>Compliant Solution</h2>
<pre>
# Some useful documentation
# ...
class configuration { ... }

# Some useful documentation
# ...
define server() { ... }
</pre>