<p>Defining several times the same key in a hash leads to unexpected behavior.</p>
<h2>Noncompliant Code Example</h2>
<pre>
$my_hash = {
  key1 => 'value 1',
  key2 => 'value 2',
  key1 => 'value 3',  # Noncompliant
}
</pre>
<h2>Compliant Solution</h2>
<pre>
$my_hash = {
  key1 => 'value 1',
  key2 => 'value 2',
  key3 => 'value 3',
}
</pre>

