<p>
    Most of the time, an empty block means that a piece of code is missing. So such empty block must be either
    filled or removed.
</p>
<p>
    This rule checks for empty:
<ul>
    <li>class or define</li>
    <li>class or define argument list</li>
    <li>resource default statement</li>
    <li>resource override</li>
    <li><code>unless</code> statement</li>
    <li><code>case</code> matcher</li>
    <li><code>if</code>, <code>elsif</code>, <code>else</code> statements</li>
</ul>
Note that no issue is raised if an empty <code>case</code> matcher, <code>elsif</code> or
<code>else</code> statement contains a comment explaining why the block is empty.
</p>

<h2>Noncompliant Code Example</h2>

<pre>
class abc {}     # Noncompliant: Remove this empty class
define abc {}    # Noncompliant: Remove this empty define

File {}                # Noncompliant: Remove this useless resource default statement
File['/tmp/log'] {}    # Noncompliant: Remove this useless resource override

if !$isValidRole {}    # Noncompliant: Remove this useless 'if' statement

if $isValidRole {
  $userRole = 'admin',
}
else {                 # Noncompliant: Empty on purpose or missing piece of code? Either remove this 'else' statement or add a comment
}

case $operatingsystem {
  'Solaris':          { include role::solaris }
  'RedHat', 'CentOS': { include role::redhat }
  default:            {}    # Noncompliant: Empty on purpose or missing piece of code? Either remove this 'default' or add a comment
}

unless $a > 5 {}    # Noncompliant: Remove this useless empty 'unless' statement
</pre>

<h2>Compliant Solution</h2>
<pre>
case $operatingsystem {
  'Solaris':          { include role::solaris }
  'RedHat', 'CentOS': { include role::redhat }
  default:            {
    # Do nothing because...
  }
}

if $isValidRole {
  $userRole = 'admin',
}
else {
  # Do nothing because...
}
</pre>
