<p>Whether the file should exist, and if so what kind of file it should be. Possible values are present, absent (also called false), file, directory, and link.</p>
<p>However, we recommend using link and target explicitly, since this behavior can be harder to read.</p>

<h2>Noncompliant Code Example</h2>

<pre>
file { '/tmp/foo':
  ensure => '/tmp/bar',
}</pre>

<h2>Compliant Solution</h2>

<pre>
file { '/tmp/foo':
  ensure => link,
  target => '/tmp/bar',
}</pre>

<h2>See</h2>
<ul>
    <li><a href="https://docs.puppetlabs.com/references/latest/type.html#file-attribute-ensure">See Puppet Labs Puppet Language File Ensure Reference</a></li>
    <li><a href="https://docs.puppetlabs.com/guides/style_guide.html#symbolic-links">See Puppet Labs Puppet Language Style Guide</a></li>
</ul>
