<p>As you are now using SonarQube, <code>lint:ignore</code> and <code>lint:endignore</code> tags can be safely removed.
</p>
<h2>Noncompliant Code Example</h2>
<pre skipautotest="">
# lint:ignore:80chars
...
# lint:endignore
</pre>

