<p>
    For readability reasons, each include should be defined on a separate line.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
include apache, haproxy
</pre>
<h2>Compliant Solution</h2>
<pre>
include apache
include haproxy
</pre>
