<p>When using <code>puppet:///</code> URLs, you should ensure that the path starts with <code>modules/</code>.
Indeed, it is the most commonly used mount point in a Puppet file server.</p>

<h2>Noncompliant Code Example</h2>
<pre>
file { '/etc/apache/apache2.conf':
  source => 'puppet:///apache/etc/apache/apache2.conf',
}
</pre>

<h2>Compliant Solution</h2>
<pre>
file { '/etc/apache/apache2.conf':
  source => 'puppet:///modules/apache2/etc/apache/apache2.conf',
}
</pre>

<h2>See</h2>
<ul>
  <li><a href="https://docs.puppetlabs.com/guides/file_serving.html#serving-module-files">The Puppet File Server Guide</a></li>
</ul>