<p>
Boolean values (true and false) behave differently when quoted ('true' and 'false'), which can lead to a fair bit of confusion. As a general rule, you should never quote booleans. This is not a style issue, rather a common mistake.
</p>

<h2>Noncompliant Code Example</h2>

<pre>
file { '/tmp/foo':
  purge => 'true',
}
</pre>

<h2>Compliant Solution</h2>

<pre>
file { '/tmp/foo':
  purge => true,
}
</pre>