<p>
When scaling user management using an external user management system is preferable to managing directly via Puppet.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
user { 'bob':
  password => 'changeme',
}
</pre>