<p>
    A chain of <code>if</code>/<code>elsif</code>/<code>else</code> or <code>case</code> or <code>selector</code> statements is evaluated from top to bottom. At most, only one branch will be executed: the first one with a condition that evaluates to <code>true</code>.
    Therefore, duplicating a condition automatically leads to dead code. Usually, this is due to a copy/paste error.
    At best, it's simply dead code and at worst, it's a bug that is likely to induce further bugs as the code is maintained, and obviously it could lead to unexpected behavior.
</p>

<h2>Noncompliant Code Example</h2>

<pre>
if $param == 1 {
  #...
} elsif $param == 2 {
  #...
} elsif $param == 1 {      # Noncompliant
  #...
}
</pre>

<h2>Compliant Solution</h2>

<pre>
if $param == 1 {
  #...
} elsif $param == 2 {
  #...
} elsif $param == 3 {
  #...
}
</pre>

<h2>See</h2>

<ul>
    <li><a href="https://www.securecoding.cert.org/confluence/x/NYA5">CERT, MSC12-C</a> - Detect and remove code that has no effect</li>
    <li><a href="https://www.securecoding.cert.org/confluence/x/SIIyAQ">CERT, MSC12-CPP</a> - Detect and remove code that has no effect</li>
</ul>
