<p>
    A chain of <code>case</code> statements is evaluated from top to bottom. At most, only one branch will be executed: the first one with a condition that evaluates to true.
    Therefore, duplicating a condition automatically leads to dead code. Usually, this is due to a copy/paste error.
    At best, it's simply dead code and at worst, it's a bug that is likely to induce further bugs as the code is maintained, and obviously it could lead to unexpected behavior.
</p>

<h2>Noncompliant Code Example</h2>

<pre>
case $param {
  5, 6: {
    ...
  }
  5: { # Noncompliant
    ...
  }
  6: { # Noncompliant
    ...
  }
  default: {
    ...
  }
}
</pre>

<h2>Compliant Solution</h2>

<pre>
case $param {
  5: {
    ...
  }
  6: {
    ...
  }
  default: {
    ...
  }
}
</pre>

<h2>See</h2>

<ul>
    <li><a href="https://www.securecoding.cert.org/confluence/x/NYA5">CERT, MSC12-C</a> - Detect and remove code that has no effect</li>
    <li><a href="https://www.securecoding.cert.org/confluence/x/SIIyAQ">CERT, MSC12-CPP</a> - Detect and remove code that has no effect</li>
</ul>