<p>
    Node names should be one of the following:

    <li>A quoted string
    <li>The bare word <code>default</code>
    <li>A regular expression</li>
</p>
<h2>Noncompliant Code Example</h2>
<pre>
node server1 {
}
</pre>
<h2>Compliant Solution</h2>
<pre>
node 'server1' {
}
</pre>
<h2>See</h2>
<ul>
    <li><a href="https://docs.puppetlabs.com/guides/style_guide.html#quoting">See Puppet Labs Puppet Language
        Style Guide</a></li>
</ul>
