<p>
  User names should not be hard coded. When defining generic modules, it should be configurable to avoid
  any unnecessary name clashes with other modules.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
user { 'tim':
  ensure => present,
}
</pre>

<h2>Compliant Solution</h2>
<pre>
user { "${username}":
  ensure => present,
}
</pre>