<?xml version="1.0" encoding="UTF-8"?>
<rules>
  <rule>
    <key>2SpaceSoftTabs</key>
    <name><![CDATA[2 Space Soft Tabs]]></name>
    <configKey>2SpaceSoftTabs</configKey>
    <description>
      <![CDATA[<p>In order to comply with the style guide, manifests must use 2 space characters when indenting.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
file { '/tmp/foo':
    ensure => present,
}</pre>

      <h2>Compliant Solution</h2>

      <pre>
file { '/tmp/foo':
  ensure => present,
}</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>HardTabs</key>
    <name><![CDATA[Hard Tabs]]></name>
    <configKey>HardTabs</configKey>
    <description>
      <![CDATA[<p>In order to comply with the style guide, manifests must not use hard tab characters (\t) in the whitespace.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
file { '/tmp/foo':
      ensure => present,
}</pre>

      <h2>Compliant Solution</h2>

      <pre>
file { '/tmp/foo':
  ensure => present,
}</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>TrailingWhitespace</key>
    <name><![CDATA[Trailing Whitespace]]></name>
    <configKey>TrailingWhitespace</configKey>
    <description>
      <![CDATA[Your manifests must not contain any trailing whitespace on any line.]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>80CharacterLineLimit</key>
    <name><![CDATA[80 Character Line Limit]]></name>
    <configKey>80CharacterLineLimit</configKey>
    <description>
      <![CDATA[Your manifests should not contain any lines longer than 80 characters.]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>ArrowAlignment</key>
    <name><![CDATA[Arrow Alignment]]></name>
    <configKey>ArrowAlignment</configKey>
    <description>
      <![CDATA[<p>Arrows (=>) within blocks of attributes (like resource declarations, selectors, hashes, etc) should be aligned with the other arrows in that block.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
file { '/tmp/foo':
  ensure => present,
  mode => '0444',
}</pre>

      <h2>Compliant Solution</h2>

      <pre>
file { '/tmp/foo':
  ensure => present,
  mode   => '0444',
}</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>SlashComment</key>
    <name><![CDATA[Slash Comment]]></name>
    <configKey>SlashComment</configKey>
    <description>
      <![CDATA[<p>Although the Puppet language allows you to use // style comments, it is recommended that you use # style comments.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
// my awesome comment
      </pre>

      <h2>Compliant Solution</h2>

      <pre>
# my awesome comment
      </pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>SlashAsteriskComment</key>
    <name><![CDATA[Slash-asterisk Comment]]></name>
    <configKey>SlashAsteriskComment</configKey>
    <description>
      <![CDATA[<p>Although the Puppet language allows you to use /* */ style multiline comments, it is recommended that you use multiple # style comments instead.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
/* my awesome comment that describes
* exactly what I'm trying to do */
      </pre>

      <h2>Compliant Solution</h2>

      <pre>
# my awesome comment that describes
  # exactly what I'm trying to do
      </pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>DoubleQuotedStrings</key>
    <name><![CDATA[Double Quoted Strings]]></name>
    <configKey>DoubleQuotedStrings</configKey>
    <description>
      <![CDATA[<p>All strings that do not contain variables or escape characters like \n or \t should be enclosed in single quotes.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
$foo = "bar"
      </pre>

      <h2>Compliant Solution</h2>

      <pre>
$foo = 'bar'
      </pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>VariablesNotEnclosed</key>
    <name><![CDATA[Variables Not Enclosed]]></name>
    <configKey>VariablesNotEnclosed</configKey>
    <description>
      <![CDATA[<p>All variables should be enclosed in in braces ({}) when being interpolated in a string.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
$foo = "bar $baz"
      </pre>

      <h2>Compliant Solution</h2>

      <pre>
$foo = "bar ${baz}"
      </pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>OnlyVariableString</key>
    <name><![CDATA[Only Variable String]]></name>
    <configKey>OnlyVariableString</configKey>
    <description>
      <![CDATA[<p>Variables standing by themselves should not be quoted. To put it another way, strings should not contain just a single variable.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
file { '/tmp/foo':
  owner => "${file_owner}",
}</pre>

      <h2>Compliant Solution</h2>

      <pre>
file { '/tmp/foo':
  owner => $file_owner,
}</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>SingleQuotedStringWithVariables</key>
    <name><![CDATA[Single Quoted String Containing Variables]]></name>
    <configKey>SingleQuotedStringWithVariables</configKey>
    <description>
      <![CDATA[<p>Single quoted strings do not get interpolated, so you should not attempt to embed variables in one. This is not a style issue, rather a common mistake.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
$foo = 'bar ${baz}'
      </pre>

      <h2>Compliant Solution</h2>

      <pre>
$foo = "bar ${baz}"
      </pre>]]>
    </description>
    <priority>CRITICAL</priority>
  </rule>
  <rule>
    <key>QuotedBooleans</key>
    <name><![CDATA[Quoted Booleans]]></name>
    <configKey>QuotedBooleans</configKey>
    <description>
      <![CDATA[<p>Boolean values (true and false) behave differently when quoted ('true' and 'false'), which can lead to a fair bit of confusion. As a general rule, you should never quote booleans. This is not a style issue, rather a common mistake.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
file { '/tmp/foo':
  purge => 'true',
}</pre>

      <h2>Compliant Solution</h2>

      <pre>
file { '/tmp/foo':
  purge => true,
}</pre>]]>
    </description>
    <priority>CRITICAL</priority>
  </rule>
  <rule>
    <key>PuppetUrlWithoutModules</key>
    <name><![CDATA[puppet:// URL without modules/]]></name>
    <configKey>PuppetUrlWithoutModules</configKey>
    <description>
      <![CDATA[<p>When using puppet:// URLs, you should ensure that the path starts with modules/ (as the most commonly used mount point in the Puppet fileserver).</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
file { '/etc/apache/apache2.conf':
  source => 'puppet:///apache/etc/apache/apache2.conf',
}</pre>

      <h2>Compliant Solution</h2>

      <pre>
file { '/etc/apache/apache2.conf':
  source => 'puppet:///modules/apache2/etc/apache/apache2.conf',
}</pre>]]>
    </description>
    <priority>CRITICAL</priority>
  </rule>
  <rule>
    <key>UnquotedResourceTitles</key>
    <name><![CDATA[Unquoted Resource Titles]]></name>
    <configKey>UnquotedResourceTitles</configKey>
    <description>
      <![CDATA[<p>All resource titles should be quoted.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
service { apache:
  ensure => running,
}</pre>

      <h2>Compliant Solution</h2>

      <pre>
service { 'apache':
  ensure => running,
}</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>AttributeOrdering</key>
    <name><![CDATA[Attribute Ordering]]></name>
    <configKey>AttributeOrdering</configKey>
    <description>
      <![CDATA[<p>If a resource declaration includes an ensure parameter, it should be the first parameter specified.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
file { '/tmp/foo':
  owner  => 'root',
  group  => 'root',
  ensure => present,
}</pre>

      <h2>Compliant Solution</h2>

      <pre>
file { '/tmp/foo':
  ensure => present,
  owner  => 'root',
  group  => 'root',
}</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>EnsureNotSymlinkTarget</key>
    <name><![CDATA[Ensure Parameter Not A Symlink Target]]></name>
    <configKey>EnsureNotSymlinkTarget</configKey>
    <description>
      <![CDATA[<p>In the interest of clarity, symbolic links should be declared by using an ensure value of ensure => link and explicitly specifying a value for the target attribute. Using a path to the target as the ensure value is not recommended.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
file { '/tmp/foo':
  ensure => '/tmp/bar',
}</pre>

        <h2>Compliant Solution</h2>

        <pre>
file { '/tmp/foo':
  ensure => link,
  target => '/tmp/bar',
}</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>FileMode</key>
    <name><![CDATA[File Modes]]></name>
    <configKey>FileMode</configKey>
    <description>
      <![CDATA[<p>File modes should be represented as 4 digits rather than 3, to explicitly show that they are octal values. File modes can also be represented symbolically e.g. u=rw,g=r.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
file { '/tmp/foo':
  mode => '666',
}</pre>

      <h2>Compliant Solution</h2>

      <pre>
file { '/tmp/foo':
  mode => '0666',
}</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>UnquotedFileMode</key>
    <name><![CDATA[Unquoted File Modes]]></name>
    <configKey>UnquotedFileMode</configKey>
    <description>
      <![CDATA[<p>File modes should be specified as single-quoted strings instead of bare word numbers.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
file { '/tmp/foo':
  mode => 0666,
}</pre>

      <h2>Compliant Solution</h2>

      <pre>
file { '/tmp/foo':
  mode => '0666',
}</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>DuplicateParameters</key>
    <name><![CDATA[Duplicate Parameters]]></name>
    <configKey>DuplicateParameters</configKey>
    <description>
      <![CDATA[<p>You really shouldn’t have duplicate parameters in a resource declaration, it is going to lead to strangness.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
file { '/tmp/foo':
  owner  => 'root',
  owner  => 'foo',
}</pre>

      <h2>Compliant Solution</h2>

      <pre>
file { '/tmp/foo':
  owner  => 'root',
}</pre>]]>
    </description>
    <priority>CRITICAL</priority>
  </rule>
  <rule>
    <key>SelectorInsideResource</key>
    <name><![CDATA[Selector Inside A Resource]]></name>
    <configKey>SelectorInsideResource</configKey>
    <description>
      <![CDATA[<p>You should not intermingle conditionals with resource declarations. When using conditionals for data assignment, you should separate conditional code from the resource declarations.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
file { '/tmp/readme.txt':
  mode => $::operatingsystem ? {
    debian => '0777',
    redhat => '0776',
    fedora => '0007',
  }
}</pre>

      <h2>Compliant Solution</h2>

      <pre>
$file_mode = $::operatingsystem ? {
  debian => '0007',
  redhat => '0776',
  fedora => '0007',
}

file { '/tmp/readme.txt':
  mode => $file_mode,
}</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>CaseWithoutDefault</key>
    <name><![CDATA[Case Without A Default Option]]></name>
    <configKey>CaseWithoutDefault</configKey>
    <description>
      <![CDATA[<p>Case statements should have default cases. Additionally, the default case should fail the catalog compilation when the resulting behavior cannot be predicted on the majority of platforms the module will be used on. If you want the default case to be “do nothing,” include it as an explicit default: {} for clarity’s sake.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
case $::operatingsystem {
  centos: {
    $version = '1.2.3'
  }
  solaris: {
    $version = '3.2.1'
  }
}</pre>

      <h2>Compliant Solution</h2>

      <pre>
case $::operatingsystem {
  centos: {
    $version = '1.2.3'
  }
  solaris: {
    $version = '3.2.1'
  }
  default: {
    fail("Module ${module_name} is not supported on ${::operatingsystem}")
  }
}</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>AutoloaderLayout</key>
    <name><![CDATA[Manifest Files In The Proper Layout]]></name>
    <configKey>AutoloaderLayout</configKey>
    <description>
      <![CDATA[<p>All classes and resource type definitions must be in separate files in the manifests directory of their module. This is functionally identical to declaring all classes and defines in init.pp, but highlights the structure of the module and makes everything more legible.</p>

      <p>Additionally, the files should be named appropriately for the class or defined type they contain. class foo should be in foo/manifests/init.pp, class foo::bar should be in foo/manifests/bar.pp and so on. You can read more about the filesystem layout for modules in the <a href="http://docs.puppetlabs.com/puppet/2.7/reference/modules_fundamentals.html#module-layout">module fundamentals documentation</a>.</p>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>RightToLeftRelationship</key>
    <name><![CDATA[Right To Left Relationship]]></name>
    <configKey>RightToLeftRelationship</configKey>
    <description>
      <![CDATA[<p>Relationship declarations with the chaining syntax should only be used in the “left to right” direction.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
Service['httpd'] <- Package['httpd']</pre>

      <h2>Compliant Solution</h2>

      <pre>
Package['httpd'] -> Service['httpd']</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>NestedClassesOrDefines</key>
    <name><![CDATA[Nested Classes Or Defines]]></name>
    <configKey>NestedClassesOrDefines</configKey>
    <description>
      <![CDATA[<p>Classes and defined resource types must not be defined within other classes.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
class apache {
  class ssl { }
}

# or

class apache {
  define config() { }
}</pre>

      <h2>Compliant Solution</h2>

      <p>Split these classes and/or defines out into seperate files as described in the <a href="http://docs.puppetlabs.com/puppet/2.7/reference/modules_fundamentals.html#module-layout">module fundamentals documentation</a>.</p>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>InheritsAcrossNamespaces</key>
    <name><![CDATA[Class Inheritance]]></name>
    <configKey>InheritsAcrossNamespaces</configKey>
    <description>
      <![CDATA[<p>Inheritance may be used within a module, but must not be used across module namespaces. Cross-module dependencies should be satisfied in a more portable way that doesn’t violate the concept of modularity, such as with include statements or relationship declarations.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
class ssh inherits server { }

class ssh::client inherits workstation { }

class wordpress inherits apache { }</pre>

      <h2>Compliant Solution</h2>

      <pre>
class ssh { }

class ssh::client inherits ssh { }

class ssh::server inherits ssh { }

class ssh::server::solaris inherits ssh::server { }</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>VariableScope</key>
    <name><![CDATA[Namespacing Variables]]></name>
    <configKey>VariableScope</configKey>
    <description>
      <![CDATA[<p>When using top-scope variables, including facts, Puppet modules should explicitly specify the empty namespace to prevent accidental scoping issues.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
$operatingsystem</pre>

      <h2>Compliant Solution</h2>

      <pre>
$::operatingsystem</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>VariableContainsDash</key>
    <name><![CDATA[Variable Format]]></name>
    <configKey>VariableContainsDash</configKey>
    <description>
      <![CDATA[<p>When defining variables you should only use letters, numbers and underscores. You should specifically not make use of dashes.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
$foo-bar</pre>

      <h2>Compliant Solution</h2>

      <pre>
$foo_bar</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>ParameterOrder</key>
    <name><![CDATA[Display Order Of Parameters]]></name>
    <configKey>ParameterOrder</configKey>
    <description>
      <![CDATA[<p>In parameterized class and defined resource type declarations, parameters that are required should be listed before optional parameters (i.e. parameters with defaults).</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
class ntp (
  $options   = "iburst",
  $servers,
  $multicast = false
) {}</pre>

      <h2>Compliant Solution</h2>

      <pre>
class ntp (
  $servers,
  $options   = "iburst",
  $multicast = false
) {}</pre>]]>
    </description>
    <priority>MINOR</priority>
  </rule>
  <rule>
    <key>ClassInheritsFromParamsClass</key>
    <name><![CDATA[Parameterised class inheritings from ‘params’ class]]></name>
    <configKey>ClassInheritsFromParamsClass</configKey>
    <description>
      <![CDATA[<p>The following popular method SHOULD NOT be used because it is not compatible with Puppet 2.6.2 and earlier.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
class ntp(
  $server = $ntp::params::server
) inherits ntp::params { }</pre>

      <h2>Compliant Solution</h2>

      <pre>
class ntp(
  $server = 'UNSET'
) {

  include ntp::params

  $server_real = $server ? {
    'UNSET' => $::ntp::params::server,
    default => $server,
  }
}</pre>]]>
    </description>
    <priority>MAJOR</priority>
  </rule>
  <rule>
    <key>NamesContainingDash</key>
    <name><![CDATA[Module Names Containing A Dash]]></name>
    <configKey>NamesContainingDash</configKey>
    <description>
      <![CDATA[<p>Support for dashes in class and defined type names differs depending on the release of Puppet you’re running. To ensure compatibility on all versions, you should avoid using dashes.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
class foo::bar-baz {}</pre>

      <h2>Compliant Solution</h2>

      <pre>
class foo::bar_baz</pre>]]>
    </description>
    <priority>MAJOR</priority>
  </rule>
  <rule>
    <key>Documentation</key>
    <name><![CDATA[Documentation]]></name>
    <configKey>Documentation</configKey>
    <description>
      <![CDATA[<p>All Puppet classes and defines should be documented via comments directly above the start of the code.</p>

      <h2>Noncompliant Code Example</h2>

      <pre>
class ntp {}</pre>

      <h2>Compliant Solution</h2>

      <pre>
# Install and configure an NTP server
# You should feel free to expand on this and document any parameters etc
class ntp {}</pre>]]>
    </description>
    <priority>MAJOR</priority>
  </rule>
  <rule>
    <key>UnquotedNodeName</key>
    <name><![CDATA[Unquoted Node Name]]></name>
    <configKey>UnquotedNodeName</configKey>
    <description>
      <![CDATA[<p>Node names should be one of the following:

      <li>A quoted string</li>
      <li>The bare word default</li>
      <li>A regular expression</li></p>

      <h2>Noncompliant Code Example</h2>

      <pre>
node server1 {
}</pre>

      <h2>Compliant Solution</h2>

      <pre>
node 'server1' {
}</pre>]]>
    </description>
    <priority>MAJOR</priority>
  </rule>
</rules>
